pkg/certificate/certificate.go

```diff
@@ -56,5 +58,18 @@ func (c *Certificate) ShouldRotate() bool {
 
 	intNoise := rand.Intn(noiseSeconds) // #nosec G404
 	secondsNoise := time.Duration(intNoise) * time.Second
-	return time.Until(c.GetExpiration()) <= (RenewBeforeCertExpires + secondsNoise)
+	renewBefore := RenewBeforeCertExpires + secondsNoise
+	// Round is called to truncate monotonic clock to the nearest second. This is done to avoid environments where the
+	// CPU clock may stop, resulting in a time measurement that differs significantly from the x509 timestamp.
+	// See https://github.com/openservicemesh/osm/issues/5000#issuecomment-1218539412 for more details.
+	expiration := c.GetExpiration().Round(0)
+	if time.Until(expiration) <= renewBefore {
+		log.Info().Msgf("Cert %s should be rotated; expires in %+v; renewBefore is %+v",
```

Comment:
you can remove the logging for the purpose of the backport, and just return time.Until...